ndmg/stats/qa_skullstrip.py

```diff
@@ -0,0 +1,244 @@
+# limitations under the License.
+#
+
+
+# Created by Vikram Chandrashekhar.
+# Edited by Greg Kiar, Eric Bridgeford and Chuankai Luo.
+# Email: Greg Kiar @ gkiar@jhu.edu   cluo16@jhu.edu
+
+import warnings
+
+warnings.simplefilter("ignore")
+import os
+import re
+import sys
+import numpy as np
+import nibabel as nb
+# import ndmg.utils as mgu
+from argparse import ArgumentParser
+from scipy import ndimage
+from matplotlib.colors import LinearSegmentedColormap
+import matplotlib as mpl
+
+mpl.use("Agg")  # very important above pyplot import
+# from nilearn.plotting.edge_detect import _edge_map as edge_map
+import matplotlib.pyplot as plt
+from ndmg.utils.qa_utils import get_min_max, opaque_colorscale,pad_im
+from ndmg.utils.gen_utils import get_filename, get_braindata
+
+
+
+
+
+def gen_overlay_pngs(
+    brain, original, outdir, loc=0, mean=False, minthr=2, maxthr=95, edge=False):
+    """
+    generate the image combine the brain and original data
+
+    Parameters
+    ----------
+    brain:
+        is the skull striped result, only left the brain
+    original:
+        is the original nii.gz file
+    outdir:
+        directory where output png file is saved
+    loc and mean is to deal with 4d data
+    minthr, maxthr, edge are the variables about the color
+    """
+    try:
+        original_name = get_filename(original)
+    except ValueError:
+        original_name = 'stripped_brain_skull_comparison'
+    brain_data = nb.load(brain).get_data()
+    if brain_data.ndim == 4:  # 4d data, so we need to reduce a dimension
+        if mean:
+            brain_data = brain_data.mean(axis=3)
+        else:
+            brain_data = brain_data[:, :, :, loc]
+    else:  # dim=3
+        brain_data = brain_data
+
+    cmap1 = LinearSegmentedColormap.from_list("mycmap1", ["white", "magenta"])
+    cmap2 = LinearSegmentedColormap.from_list("mycmap2", ["white", "green"])
+
+
+    fig = plot_overlays_skullstrip(brain_data, original, [cmap1, cmap2], minthr, maxthr, edge)
+
+    # name and save the file
+    fig.savefig(f"{outdir}/qa_skullstrip__{original_name}.png", format="png")
+
+
+
+
+
+def plot_overlays_skullstrip(b0, original, cmaps=None, minthr=2, maxthr=95, edge=False):
+    """
+    generate the image combine the brain and original data
+
+    Parameters
+    ----------
+    b0:
+        is the skull striped result, only left the brain
+    original:
+        is the original nii.gz file
+    cmaps, minthr, maxthr, edge are the variables about the color
+    """
+
+    plt.rcParams.update({"axes.labelsize": "x-large", "axes.titlesize": "x-large"})
+    foverlay = plt.figure()
+
+    original = get_braindata(original)
+    ori_shape = get_braindata(b0).shape
+    b0 = get_braindata(b0)
+    if original.shape != b0.shape:
+        raise ValueError("Two files are not the same shape.")
+    b0 = pad_im(b0, max(ori_shape[0:3]), 0, False)
+    original = pad_im(original,max(ori_shape[0:3]),0, False)
```

Comment:
for `pad_im`, I told Wilson to make `0` and `False` keyword arguments. When he does that, and you merge with his code, it might break this code. Please make sure to talk to him and update this if it'll cause problems.

Reply:
will fix it when Wilson is ready

Reply:
Wilson is ready

Reply:
https://github.com/neurodata/ndmg/pull/366/commits/19a993b9f5c2c92ce68c5e335bf0dc384f69651a